### PR TITLE
make input paths relative to current working directory

### PR DESCRIPTION
**Pull request by brekk**
Thanks for this library, it's exactly what I was looking for!

I was unable to run this library without augmenting it:

```sh
tscodeshift -t dist/src/transforms/mocha.js 'my-fixture-file.ts'
```

Always resulted in:

```
Failed to run tscodeshift. {
    Error: Cannot find module 'dist/src/transforms/mocha.js'
    [...]
}
```

*node version*: `v8.5.0`

I also added a test to prove that this works as expected.

Please let me know if there are additional things you'd like me to change.

### CODE REVIEW

**Comment by KnisterPeter** [comment]
Awesome. Thanks for the contribution. :rocket: 
After a first look, I would like to have the yarn file removed. I favor npm and even without yarn.lock it will be installable with yarn. The yarn.lock file will be too quickly out of date.

**Comment by KnisterPeter** [comment]
For relative paths you could also used 'tscodeshift -t ./dist/src/transforms/mocha.js 'my-fixture-file.ts'`
The `./` in front of the relative path makes it a node file and not a node module.

**Comment by KnisterPeter** [comment]
And since you updated typescript to the latest version it my be required to update the test harness as well.
This is a manuall npm script and maybe a few parser updates to implement.
The harness is cached and only runs manually and on CI for performance reasons.

EDIT: Oh, don't mind. This would be located in the `ts-emitter` project. Not in here. So this is fine.

**Comment by brekk** [comment]
@KnisterPeter Cool, I'll remove the `yarn.lock` file. I could be wrong here but I believe I tried it with `./` relative pathing and had the same issue -- will verify.

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "start": "npm test -- --watch",
     "clean": "rimraf dist",
     "build": "tsc",
-    "test": "jest",
+    "test": "npm run build && jest",
```
Comment (KnisterPeter) [inline package.json:16]: Running build before jest is not required with ts-jest.

File: package.json [hunk 2]
```diff
@@ -59,13 +60,16 @@
     "transform": {
       ".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"
     },
-    "testRegex": "\\.test\\.(ts|tsx)$",
+    "testRegex": "\\.test\\.(ts|tsx|js)$",
```
Comment (KnisterPeter) [inline package.json]: Why do you need to find js files for tests?

File: src/index.test.ts [hunk 3]
```diff
@@ -123,3 +123,68 @@ it('convert var declarations to let declarations', () => {
 
   expect(actual).toBe(expected);
 });
+
+test(`api`, () => {
```
Comment (KnisterPeter) [inline src/index.test.ts]: I'm not sure what you are testing here.

File: relative-paths.test.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/* global test, expect, afterEach */
```
Comment (KnisterPeter) [inline relative-paths.test.js]: This files belongs into the src folder. Even tests are sources.

File: package.json (hunk 1 above)
Comment (brekk) [reply package.json:16]: So, relative to how I am building stuff, this is; basically I wanted to be able to ask the compiled / for-publication version to build things at the command-line, and `dist/src` doesn't exist at that point.

More comments below.

File: relative-paths.test.js (hunk 4 above)
Comment (brekk) [reply relative-paths.test.js]: So, in general the reason I have this `.js` file is more from my lack of familiarity with TypeScript. I am more than happy to have this file be in `src`, but relative to the problem I was trying to solve, I didn't wanna jump through hoops of TypeScript fixing while trying to get the gist of this thing working. I think specifically I was running into some issues trying to use `path` -- this evening I'll try and regroup and clarify my intentions in this PR.

File: src/index.test.ts (hunk 3 above)
Comment (brekk) [reply src/index.test.ts]: So, there's an else case that is very hard to get to inside of the `./transform#api.tscodeshift` function, and this test covers that else case. Sorry, I should have clarified my test case after I got it working. (Basically the coverage was lowered by me adding the files I added, so by covering the else case, I was able to get coveralls to pass.)

File: src/transform.ts [hunk 5]
```diff
@@ -2,8 +2,8 @@ import * as ts from 'typescript';
 import { Collection } from './collection';
 import * as types from './types';
 
-const api = {
-  tscodeshift: (source: string|ts.Node) => {
+export const api = {
+  tscodeshift: (source: string|ts.Node): Collection<any, any> => {
```
Comment (brekk) [inline src/transform.ts:6]: @KnisterPeter I had to make this change in order to get `tsc` to compile properly -- I'm guessing there's a better way to do this, but I was running into some problems locally.

File: relative-paths.test.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+const execa = require('execa')
+const path = require('path')
+
+const CLI = path.resolve(__dirname, `./dist/src/index.js`)
```
Comment (brekk) [inline relative-paths.test.js]: @KnisterPeter This is the reason for that additional build step -- I want to be able to reference the compiled version of this file in the test.

File: package.json (hunk 2 above)
Comment (brekk) [reply package.json]: See comment [here](https://github.com/KnisterPeter/tscodeshift/pull/20#discussion_r142551695)

File: src/index.test.ts (hunk 3 above)
Comment (brekk) [reply src/index.test.ts]: Specifically this: https://travis-ci.org/KnisterPeter/tscodeshift/builds/237974069#L217

File: src/index.test.ts [hunk 7]
```diff
@@ -124,7 +124,7 @@ it('convert var declarations to let declarations', () => {
   expect(actual).toBe(expected);
 });
 
-test(`api`, () => {
+test(`api.tscodeshift can process raw ts.Nodes`, () => {
   expect(typeof api).toBe(`object`);
   expect(typeof api.tscodeshift).toBe(`function`);
   const removeCircularCrap = (x: any): any => {
```
Comment (KnisterPeter) [inline src/index.test.ts]: Rename this from 'Crap' to 'References' :wink:

File: src/index.test.ts [hunk 8]
```diff
@@ -136,7 +136,6 @@ test(`api`, () => {
     return x;
   };
   const expected = JSON.parse(JSON.stringify(removeCircularCrap(api.tscodeshift(`false`))));
-  // expect(JSON.parse(JSON.stringify(removeCircularCrap(expected)))).toEqual(circular)
```
Comment (KnisterPeter) [inline src/index.test.ts]: Remove commented out lines

File: src/relative-paths.test.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
+/* global test, expect, afterEach */
+// https://stackoverflow.com/questions/39415661
```
Comment (KnisterPeter) [inline src/relative-paths.test.ts]: You do not need this reference for the import-require statement.
Its part of the official typescript documentation. No need to quote these.

https://www.typescriptlang.org/docs/handbook/modules.html

**Comment by KnisterPeter** [comment]
Besides all my comments. Thanks for you awesome contribution. :+1:

**Comment by KnisterPeter** [comment]
I've updated your test code to do not require `npm run build` before `jest`.
This means `npm run jest -- --watch` is possible again.

**Pull request merged**